microcosm_fastapi/database/store.py

```diff
@@ -164,8 +164,9 @@ async def count(self, *criterion, **kwargs):
         """
         Count the number of models matching some criterion.
         """
-        query = select(func.count(self.model_class.id))
-        query = self._where(query, **kwargs)
+        query = self._query(*criterion)
+        query = self._where(query, **kwargs).subquery()
+        query = select(func.count(query.c.id))
```

Comment:
One of my test cases was failing as it didn't have the sufficient model_class information to be able to effectively do its a query when we do `select(func.count(self.model_class.id))` (i.e we only have the model_class.id available when we might need other properties...)

I've shifted it round so the func.count comes last and doesn't disruped the self._where which might need more model_class properties than just the model_class.id